docs/alerts.md

```diff
@@ -0,0 +1,183 @@
+# Alert Handling
+
+## Overview
+
+### Requirements
+
+The following are considered requirements for the alert package to both support and/or implement.
+
+* It must be able to consume data from an arbitrary query and evaluate a level function on each row in each table.
+* It must be possible to store this result in a system bucket called a `status bucket`.
+* It must be able to evaluate a `notification rule` based on the output of the previous or from the data stored in the status bucket.
+* The notification must send messages to third party endpoints when the rule indicates.
+* The notification events must be capable of being logged into a `notification bucket`.
+
+### Query Lifecycle
+
+The query lifecycle is tangentially related to alerts.
+At the moment, the flux engine is currently only used for running batch requests.
+This means that there is only one table for each group key that goes through the pipeline.
+
+The alert package as below does not assume this.
+It is designed so the functions will work regardless of whether run in batch mode or if the query were being run with each new row triggering a new table.
+For ease of reading, the example tables will be shown as if only one table were shown.
```

Comment:
I suppose that the most common use case for alerting should be to embed it into a task.

The problem with tasks is that you cannot share data between many invocations of the batch job. That would require real streaming. I don't know if we need that capability at the moment or in the future.

Reply:
@nathanielc mentioned in the connected issue that this was how we were going to do alerting. I don't know if that's permanent or just at the moment until we figure out a way to run queries continuously, but I think the model outlined here will work regardless of the mode.

Reply:
Using tasks as the backbone of alerting is the long term plan, its not temporary.

Reply:
I more meant if tasks would be repeatedly executed or we would have tasks be something that is continuously run. If we don't plan to have any streaming functionality, I think we may need to have a conversation about triggers as the removal of triggers could simplify the query engine a decent amount.

Reply:
We do not plan to have a true streaming query any time soon. Lets have the discussion on triggers separate from the alert/notification design. We will revisit the alerts design when we go to introduce true streaming later along with many other designs.